### PR TITLE
Ensure there is not several certificates with same subject in truststore

### PR DESCRIPTION
**Pull request by sbernard31**
(link to #1307)

### CODE REVIEW

File: scandium-core/src/main/java/org/eclipse/californium/scandium/config/DtlsConnectorConfig.java [hunk 1]
```diff
@@ -2487,5 +2488,17 @@ private void determineCipherSuitesFromConfig() {
 
 			config.supportedCipherSuites = ciphers;
 		}
+
+		private void checkTrustStore() {
+			X509Certificate[] store = config.trustStore;
+			for (int i = 0; i < store.length; i++) {
```
Comment (boaks) [inline scandium-core/src/main/java/org/eclipse/californium/scandium/config/DtlsConnectorConfig.java:2494]: What's about using `CertPathUtil.toSubjects` and `List.contains`?

**Comment by boaks** [comment]
As I wrote in [my comment](https://github.com/eclipse/californium/issues/1307#issuecomment-624718489), I'm not sure about the rules, which applies to x509.

If you think, that is worth to be checked, go for it.
Maybe you replace the nested for-loops by one loop and a `List.contains`.

```
while (principal = list.remove(0)) {
 if (list.contains(principal)) {
 }
}
```

**Comment by sbernard31** [comment]
Let's think and discuss a bit more about that in #1307

**Comment by sbernard31** [comment]
(I suspect this will need more discussion than expected so maybe we can let this PR aside for the 2.2.3)

**Comment by boaks** [comment]
You may add it, I don't object that.
I would prefer to add the javadoc I recommend in the issue comment.

**Comment by boaks** [comment]
LGTM.
(squash before merge?)

**Comment by boaks** [comment]
> (squash before merge?)

Uups, I mixed up the number of files and commits.

**Comment by sbernard31** [comment]
(integrated in master too 3e5bb4d)

**Pull request merged**